error/s2n_errno.c

```diff
@@ -185,6 +185,13 @@ typedef union {
 
 static struct s2n_error_translation *s2n_lookup_error_translation(int error)
 {
+    if (error_translation_table == NULL) {
+        /* since error_translation_table should be initialized
+         * early in s2n_init(), this should almost never happen,
+         * unless an error is thrown before, i.e. in s2n_fips_init()
+         */
+        return NULL;
```

Comment:
I think we might need a failover here in order to get the error translation here instead of failing the call. Since we might end up here for 2 corner cases:
1. s2n_fips_init fails
2. fail to allocate memory for hash table

I suggest we fallback to O(N) alogrithm - iterate error arrary https://github.com/awslabs/s2n/commit/c75d57d89bfd3e7b6f3fbdaed91c726f3db8fad2#diff-4b946e7f1437f8cbd0865259cbe20aaeL177